runtime/gc_include/ObjectAccessBarrierAPI.hpp

```diff
@@ -2675,10 +2643,13 @@ class MM_ObjectAccessBarrierAPI
 	internalConvertTokenFromPointer(j9object_t pointer)
 	{
 #if defined (OMR_GC_COMPRESSED_POINTERS)
-		return convertTokenFromPointer(pointer, _compressedPointersShift);
-#else /* OMR_GC_COMPRESSED_POINTERS */
-		return (fj9object_t)pointer;
+		if (compressObjectReferences()) {
+			return (fj9object_t)((UDATA)pointer >> _compressedPointersShift);
+		} else
 #endif /* OMR_GC_COMPRESSED_POINTERS */
+		{
+			return (fj9object_t)(UDATA)pointer;
+		}
```

Comment:
This itself is good, but similarly to slow barrier code, this is invoked from a site where there is already a runtime CR check (see readObjectImpl).

As mentioned before, we could specialize this method further into two methods for CR and nonCR, and use proper one from the call site that already knows if it's CR or not.